pkg/redis/redis.go

```diff
@@ -78,15 +78,29 @@ type Config struct {
 	Namespace []string
 }
 
+// NewRedisClient connects with the correct client type to Redis
+func NewRedisClient(conf config.Redis) *redis.Client {
+	if conf.EnableFailover {
+		return redis.NewFailoverClient(&redis.FailoverOptions{
+			MasterName:    conf.FailoverMasterName,
```

Comment:
But there's no error checking if this is empty, right?

Shouldn't we fail if failover is set to true but a master name is not defined? That is now silently considered `IsZero()`

Reply:
If the user enables failover but has invalid (or missing) arguments, then the client creation fails:

```
$ go run ./cmd/ttn-lw-stack start --redis.enable-failover=true  
  INFO Setting up core component               
(.........)        
  INFO Setting up Network Server               
error:cmd/internal/shared:initialize_network_server (could not initialize Network Server)
--- error:pkg/redis:store (store error)
--- redis: all sentinels are unreachable
exit status 255
```

This is consistent with how the stack fails to initiate when Redis is unavailable:

```
$ go run ./cmd/ttn-lw-stack start --redis.address=localhost:16379
  INFO Setting up core component    
(.........) 
  INFO Setting up Network Server               
error:cmd/internal/shared:initialize_network_server (could not initialize Network Server)
--- error:pkg/redis:store (store error)
--- dial tcp 127.0.0.1:16379: connect: connection refused
exit status 255
```

Is this the desired behaviour?

Reply:
Yes